From patchwork Mon Jul 10 00:19:50 2017
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
Subject: [LEDE-DEV,
 3/4] scripts/getver.sh: include branch name in revision string
X-Patchwork-Submitter: Matthias Schiffer <mschiffer@universe-factory.net>
X-Patchwork-Id: 785975
X-Patchwork-Delegate: mschiffer+lede@universe-factory.net
Message-Id: <d55f4b29e90d5cf78d2f3472f3cc774f9058445b.1499645991.git.mschiffer@universe-factory.net>
To: lede-dev@lists.infradead.org,
	zajec5@gmail.com
Date: Mon, 10 Jul 2017 02:19:50 +0200
From: Matthias Schiffer <mschiffer@universe-factory.net>
List-Id: <lede-dev.lists.infradead.org>

To determine the originating branch of a commit, a file called "branch"
must be added to the root of the repository (similar to the existing
"version" file used to override the getver.sh logic). When no such file
exists in the commit in question, the master branch is assumed.

Signed-off-by: Matthias Schiffer <mschiffer@universe-factory.net>
---
 scripts/getver.sh | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/scripts/getver.sh b/scripts/getver.sh
index 215bb036a5..f6d18bb3a3 100755
--- a/scripts/getver.sh
+++ b/scripts/getver.sh
@@ -5,6 +5,10 @@ export LC_ALL=C
 
 GET_REV=$1
 
+get_branch() {
+	git show "$1:branch" 2>/dev/null || echo 'master'
+}
+
 try_version() {
 	[ -f version ] || return 1
 	REV="$(cat version)"
@@ -42,7 +46,7 @@ try_git() {
 			REV="${UPSTREAM_REV}+$((REV - UPSTREAM_REV))"
 		fi
 
-		REV="${REV:+r$REV-$(git log -n 1 --format="%h" $UPSTREAM_BASE)}"
+		REV="$(get_branch "$GET_REV")-${REV:+r$REV-$(git log -n 1 --format="%h" $UPSTREAM_BASE)}"
 
 		;;
 	esac
